source/interface/transport_interface.h

```diff
@@ -96,16 +96,31 @@
  *                                      size_t bytesToRecv )
  * {
  *     int32_t bytesReceived = 0;
- *     bytesReceived = TLSRecv( pNetworkContext->tlsContext,
- *                              pBuffer,
- *                              bytesToRecv,
- *                              MY_SOCKET_TIMEOUT );
- *     if( bytesReceived < 0 )
+ *     bool callTlsRecvFunc = true;
+ *
+ *     // For a single byte read request, check if data is available on the network.
+ *     if( bytesToRecv == 1 )
  *     {
- *         // Handle socket error.
+ *        // If no data is available on the network, do not call TLSRecv
+ *        // to avoid blocking for socket timeout.
+ *        if( TLSRecvCount( pNetworkContext->tlsContext ) == 0 )
+ *        {
+ *            callTlsRecvFunc = false;
+ *        }
  *     }
- *     // Handle other cases.
  *
+ *     if( callTlsRecvFunc == true )
+ *     {
+ *        bytesReceived = TLSRecv( pNetworkContext->tlsContext,
+ *                                 pBuffer,
+ *                                 bytesToRecv,
+ *                                 MY_SOCKET_TIMEOUT );
+ *        if( bytesReceived < 0 )
+ *        {
+ *            // Handle socket error.
```

Comment:
Here if bytesReceived contains an error code that just means the TCP read timed (or would block) then that is not an error so bytesReceived needs to be translated into whatever the calling function will interpret as "try again" - which for our code is zero, but if this function is calling the TCP read function and passing a value back to the TLS stack then zero may also be interpreted as an error.

Reply:
Have updated the pseudo code block to add a check for a retriable error.